### PR TITLE
Auth: Add organization mapping configuration to the UI

### PR DESCRIPTION
**Pull request by mgyongyosi**
**What is this feature?**
1. This PR adds the capability to configure the `org_mapping` and the `org_attribute_path` for the OAuth providers.
1. Fixes parsing of config stored in JSON array format and it changes the default behaviour for storing arrays: from now on arrays are stored in JSON array format if the array contains multiple values.

**Why do we need this feature?**
- Makes configuring org_mapping and org_attribute_path from the UI possible.
- Fixes parsing of configuration stored in JSON array format.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/ssosettings/validation/oauth_validators.go [hunk 1]
```diff
@@ -17,6 +17,20 @@ func AllowAssignGrafanaAdminValidator(info *social.OAuthInfo, requester identity
 	return nil
 }
 
+func OrgMappingValidator(info *social.OAuthInfo, requester identity.Requester) error {
+	if len(info.OrgMapping) > 0 && !requester.GetIsGrafanaAdmin() {
```
Comment (dmihai) [inline pkg/services/ssosettings/validation/oauth_validators.go:21]: If `orgMapping` is configured for a provider then only Grafana Admins will be able to edit the configuration. What I think we can do instead is to check whether non Grafana admins are modifying the existing `orgMapping`. The SSO validation function contains the `oldSettings` parameter so we can check if `oldSettings.orgMapping` is the same with `currentSettings.orgMapping`.

File: pkg/services/ssosettings/validation/oauth_validators.go [hunk 2]
```diff
@@ -17,6 +17,20 @@ func AllowAssignGrafanaAdminValidator(info *social.OAuthInfo, requester identity
...
+		return ssosettings.ErrInvalidOAuthConfig("Organization mapping can only be updated by Grafana Server Admins.")
+	}
+	return nil
+}
+
+func OrgAttributePathValidator(info *social.OAuthInfo, requester identity.Requester) error {
+	if info.OrgAttributePath != "" && !requester.GetIsGrafanaAdmin() {
```
Comment (dmihai) [inline pkg/services/ssosettings/validation/oauth_validators.go:28]: Similar to my previous comment, I think we should better check if `OrgAttributePath` has been modified by non Grafana admins.

File: public/app/features/auth-config/fields.tsx [hunk 3]
```diff
@@ -448,6 +457,22 @@ export function fieldMap(provider: string): Record<string, FieldData> {
       description: 'Prevent synchronizing users’ organization roles from your IdP.',
       type: 'switch',
     },
+    orgMapping: {
+      label: 'Organization mapping',
+      description: orgMappingDescription(provider),
+      type: 'select',
+      hidden: !contextSrv.isGrafanaAdmin,
```
Comment (dmihai) [inline public/app/features/auth-config/fields.tsx:464]: If this field is hidden then its current value will be sent to the API on submit? I think we want to pass the existing value for this field in case the user is not a Grafana admin.

Comment (mgyongyosi) [reply public/app/features/auth-config/fields.tsx:464]: Yes, this is a similar problem we have with AllowAssignGrafanaAdmin and wanted to discuss with the team!

File: public/app/features/auth-config/ProviderConfigForm.test.tsx [hunk 4]
```diff
@@ -3,6 +3,7 @@ import userEvent from '@testing-library/user-event';
 import { JSX } from 'react';
 
 import { reportInteraction } from '@grafana/runtime';
+// import { contextSrv } from 'app/core/core';
```
Comment (dmihai) [inline public/app/features/auth-config/ProviderConfigForm.test.tsx:6]: Can this be removed?

File: pkg/login/social/connectors/signature.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
+package connectors
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+
+	"github.com/go-jose/go-jose/v3"
+)
+
+func SignToken(data []byte, privateKey *ecdsa.PrivateKey) (string, error) {
```
Comment (dmihai) [inline pkg/login/social/connectors/signature.go:10]: I don't see this function being used in this PR. Should this be part of a separate PR?

Comment (mgyongyosi) [reply pkg/login/social/connectors/signature.go:10]: Yep, it's been added accidentally.

File: pkg/services/ssosettings/validation/oauth_validators.go (hunk 1 above)
Comment (mgyongyosi) [reply pkg/services/ssosettings/validation/oauth_validators.go]: Done

File: public/app/features/auth-config/fields.tsx (hunk 3 above)
Comment (mgyongyosi) [reply public/app/features/auth-config/fields.tsx:464]: Done

File: pkg/login/social/connectors/signature.go (hunk 5 above)
Comment (mgyongyosi) [reply pkg/login/social/connectors/signature.go]: Removed

**Review by dmihai — APPROVED** [review]
